packages/babel-plugin/src/class-names/index.tsx

```diff
@@ -13,4 +63,55 @@ import { State } from '../types';
  * @param path Babel path - expects to be a JSX opening element.
  * @param state Babel state - should house options and meta data used during the transformation.
  */
-export const visitClassNamesPath = (_: NodePath<t.JSXOpeningElement>, __: State) => {};
+export const visitClassNamesPath = (path: NodePath<t.JSXElement>, meta: Metadata) => {
+  if (
+    t.isJSXIdentifier(path.node.openingElement.name) &&
+    path.node.openingElement.name.name !== meta.state.compiledImports?.ClassNames
+  ) {
+    // We aren't interested in this element. Bail out!
+    return;
+  }
+
+  const children = getJsxChildrenAsFunction(path);
+  let variables: CSSOutput['variables'] = [];
+  let sheets: string[] = [];
+
+  // First pass to replace all usages of `css({})`
+  path.traverse({
+    Expression(path) {
+      const styles = extractStyles(path);
+      if (!styles) {
+        // Nothing to do - skip.
+        return;
+      }
+
+      const builtCss = buildCss(styles, meta);
```

Comment:
Nice. API's are getting simpler now. Reusing existing utilities 😄

Reply:
yeah, amusingly not much code needed